src/printer/index.js

```diff
@@ -878,12 +878,14 @@ function genericPrint(path, options, print) {
     }
 
     case "comprehension": {
-      const parts = [printForIn(path, print)];
+      let parts = [printForIn(path, print)];
 
       if (n.ifs.length > 0) {
-        parts.push(
-          line,
-          groupConcat(["if", line, join(line, path.map(print, "ifs"))])
+        parts = parts.concat(
+          path.map(
+            node => concat([line, groupConcat(["if", line, node.call(print)])]),
```

Comment:
nit: `path` would probably be a variable name than `node` since the object is actually the same `FastPath` instance that you're calling `.map` on. Plus, in other parts of the code I've seen `node` or `n` refer to the actual AST node object which could make this extra confusing haha.